### PR TITLE
Replace TSLint with ESLint

### LINKED ISSUE #2882: TSLint is deprecated
https://github.com/palantir/tslint/issues/4534

### PR DESCRIPTION
**Pull request by andschwa**
The originally recommended migration method resulted in 3,150 problems, and it moved us to an already out-of-date version of ESLint. So I decided to redo this by moving natively to ESLint's recommended configuration (no TSLint migration, except for comment directives) and then allowed ESLint to reformat the codebase. Resolves #2882.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -64,6 +64,13 @@
     "@types/sinon": "~10.0.0",
     "@types/uuid": "~8.3.0",
     "@types/vscode": "~1.52.0",
+    "@typescript-eslint/eslint-plugin": "^4.22.0",
+    "@typescript-eslint/eslint-plugin-tslint": "^4.22.0",
+    "@typescript-eslint/parser": "^4.22.0",
+    "eslint": "^7.25.0",
```
Comment (andschwa) [inline package.json:70]: I did manually update this to 7.25 since the converter brings in 6.x, but even before that, there were thousands of errors.

**Comment by andschwa** [comment]
@SeeminglyScience for what it's worth there is also `@typescript-eslint/recommended-requiring-type-checking` which would be _awesome_ to have...but currently finds over 600 errors.

**Comment by andschwa** [comment]
Followed this essentially: https://typescript-eslint.io/docs/linting/configs/

**Review by SeeminglyScience — APPROVED** [review]
LGTM!

**Pull request merged**